src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Calendar/Calendar.shared.cs

```diff
@@ -0,0 +1,533 @@
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+
+			Grid.SetRow(gridWeekDayHeaders, 0);
+			layoutRoot.Children.Add(gridWeekDayHeaders);
+			this.gridWeekDayHeaders = gridWeekDayHeaders;
+
+			var gridDays = new Grid
+			{
+				ColumnSpacing = 0
+			};
+
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+
+			Grid.SetRow(gridDays, 1);
+			layoutRoot.Children.Add(gridDays);
+			this.gridDays = gridDays;
+
+			Content = layoutRoot;
+		}
+
+		void UpdateCalendarDays()
+		{
+			UpdateGridForDays();
+			UpdateDays();
+			SelectDays(SelectedDays);
+		}
+
+		void SelectDays(IEnumerable<DateTime> daysSelected)
+		{
+			foreach (var day in days)
+			{
+				var isSelected = daysSelected.Any(x => x.Date == day.Date);
+				day.IsSelected = isSelected;
+			}
+		}
+
+		void UpdateGridForDays()
+		{
+			if (!ShowWeekends)
+			{
+				gridDays.ColumnDefinitions[5].Width = new GridLength(0);
+				gridDays.ColumnDefinitions[6].Width = new GridLength(0);
+			}
+			else if (ShowWeekends)
+			{
+				gridDays.ColumnDefinitions[5].Width = GridLength.Star;
+				gridDays.ColumnDefinitions[6].Width = GridLength.Star;
+			}
+
+			var weeksInMonth = Date.WeeksInMonth(FirstDayOfWeek);
+
+			if (gridDays.RowDefinitions.Count < weeksInMonth)
+			{
+				for (var row = gridDays.RowDefinitions.Count; row < weeksInMonth; row++)
+				{
+					gridDays.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+					var daysToAdd = days.Where(x => Grid.GetRow(x) == row).ToList();
+
+					foreach (var calendarDay in daysToAdd)
+					{
+						gridDays.Children.Add(calendarDay);
+					}
+				}
+			}
+			else if (gridDays.RowDefinitions.Count > weeksInMonth)
+			{
+				for (var row = gridDays.RowDefinitions.Count; row > weeksInMonth; row--)
+				{
+					gridDays.RowDefinitions.RemoveAt(gridDays.RowDefinitions.Count - 1);
+
+					var daysToRemove = days.Where(x => Grid.GetRow(x) == row - 1).ToList();
+
+					foreach (var calendarDay in daysToRemove)
+					{
+						gridDays.Children.Remove(calendarDay);
+					}
+				}
+			}
```

Comment:
Here we can cache the `gridDays.RowDefinitions.Count` avoiding to calling the `get` method everywhere. If I'm correct that value will not change inside this method.
```suggestion			
                        var gridDaysRowDefinitionsCount = gridDays.RowDefinitions.Count;
			if (gridDaysRowDefinitionsCount < weeksInMonth)
			{
				for (var row = gridDaysRowDefinitionsCount; row < weeksInMonth; row++)
				{
					gridDays.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });

					var daysToAdd = days.Where(x => Grid.GetRow(x) == row).ToList();

					foreach (var calendarDay in daysToAdd)
					{
						gridDays.Children.Add(calendarDay);
					}
				}
			}
			else if (gridDaysRowDefinitionsCount > weeksInMonth)
			{
				for (var row = gridDaysRowDefinitionsCount; row > weeksInMonth; row--)
				{
					gridDays.RowDefinitions.RemoveAt(gridDaysRowDefinitionsCount - 1);

					var daysToRemove = days.Where(x => Grid.GetRow(x) == row - 1).ToList();

					foreach (var calendarDay in daysToRemove)
					{
						gridDays.Children.Remove(calendarDay);
					}
				}
			}
```

Reply:
thanks, I've added caching